upstash/vector/index/crud.go

```diff
@@ -0,0 +1,119 @@
+package index
+
+import (
+	"context"
+	"github.com/upstash/terraform-provider-upstash/upstash/utils"
+
+	"github.com/upstash/terraform-provider-upstash/upstash/client"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	_ "github.com/upstash/terraform-provider-upstash/upstash/client"
+	_ "github.com/upstash/terraform-provider-upstash/upstash/utils"
+)
+
+func resourceIndexCreate(ctx context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.UpstashClient)
+
+	index, err := CreateIndex(c, CreateIndexRequest{
+		Name:               data.Get("name").(string),
+		SimilarityFunction: data.Get("similarity_function").(string),
+		DimensionCount:     data.Get("dimension_count").(int),
+		Region:             data.Get("region").(string),
+		Type:               data.Get("type").(string),
+	})
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(index.Id)
+
+	err = data.Set("id", index.Id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceIndexRead(ctx, data, m)
+
+}
+func resourceIndexRead(ctx context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
More like a consistency thing, but can you add a newline where a function ends and new one starts? Also, some functions have newlines at the very end, and some don't. Maybe let's uniformize this :D